public/utils/datasource-utils.ts

```diff
@@ -27,7 +27,12 @@ export function getClusterInfo(dataSourceEnabled: boolean, cluster: DataSourceOp
 export function getDataSourceFromUrl(): DataSourceOption {
   const urlParams = new URLSearchParams(window.location.search);
   const dataSourceParam = (urlParams && urlParams.get(DATASOURCEURLKEY)) || '{}';
```

Comment:
nit: Why no explicitly check if `urlParams.has(DATASOURCEURLKEY)` here and return empty object/null/undefined to indicate to the caller that the request does not contain dataSource?

~~return JSON.parse('{}');~~ `return {};`

Reply:
the urlParam needs to be parsed to a valid JSON entity. If we directly return without proper parsing, the page will crash, which is what I'm trying to solve with this change.